zebrad/src/components/mempool/gossip.rs

```diff
@@ -37,14 +37,35 @@ where
     let mut broadcast_network = Timeout::new(broadcast_network, TIPS_RESPONSE_TIMEOUT);
 
     loop {
-        // once we get new data in the channel, broadcast to peers
+        let mut combined_changes = 1;
+
+        // once we get new data in the channel, broadcast to peers,
+        // the mempool automatically combines some transactions that arrive close together
         receiver.changed().await?;
+        let mut txs = receiver.borrow().clone();
+        tokio::task::yield_now().await;
+
+        // also combine transactions that arrived shortly after this one
+        while receiver.has_changed()? && combined_changes < MAX_CHANGES_BEFORE_SEND {
```

Comment:
```suggestion
        while receiver.has_changed()? && combined_changes <= MAX_CHANGES_BEFORE_SEND {
```

Do we want to loop 10 times or 9 ?

Reply:
I don't think it matters either way. We have already checked once before the loop, so it's 10 combined changes at the end of the loop.